ci-operator/step-registry/cucushift/installer/check/vsphere/invalid-fields/cucushift-installer-check-vsphere-invalid-fields-commands.sh

```diff
@@ -97,7 +96,210 @@ else
   echo "Fail: failed to check cluster name"
   check_result=$((check_result + 1))
 fi
-set -o errexit
+rm -rf "${dir:?}/"
+mkdir "${dir}/"
+
+# static ip, no bootstrap role
+cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
+CONFIG="${dir}/install-config.yaml"
+yq-go d -i "${CONFIG}" "platform.vsphere.hosts.(role==bootstrap)"
+run_install
+if grep "a single host with the bootstrap role must be defined" "${dir}"/.openshift_install.log; then
+  echo "Pass: passed to check bootstrap role"
+else
+  echo "Fail: failed to check bootstrap role"
+  check_result=$((check_result + 1))
+fi
+rm -rf "${dir:?}/"
+mkdir "${dir}/"
+
+# static ip, no control-plane role
+cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
+CONFIG="${dir}/install-config.yaml"
+yq-go d -i "${CONFIG}" "platform.vsphere.hosts.(role==control-plane)"
+run_install
+if grep "not enough hosts found (0) to support all the configured control plane replicas (3)" "${dir}"/.openshift_install.log; then
+  echo "Pass: passed to check control-plane role"
+else
+  echo "Fail: failed to check control-plane role"
+  check_result=$((check_result + 1))
+fi
+rm -rf "${dir:?}/"
+mkdir "${dir}/"
+
+# static ip, no compute role
+cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
+CONFIG="${dir}/install-config.yaml"
+yq-go d -i "${CONFIG}" "platform.vsphere.hosts.(role==compute)"
+run_install
+if grep "not enough hosts found (0) to support all the configured compute replicas" "${dir}"/.openshift_install.log; then
+  echo "Pass: passed to check compute role"
+else
+  echo "Fail: failed to check compute role"
+  check_result=$((check_result + 1))
+fi
+rm -rf "${dir:?}/"
+mkdir "${dir}/"
+
+# static ip, invalid role name
+cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
+CONFIG="${dir}/install-config.yaml"
+PATCH="${dir}/invalid-role-name.yaml.patch"
+cat >"${PATCH}" <<EOF
+platform:
+  vsphere:
+    hosts:
+      - role: bootstrap-invalid
+      - role: control-plane-invalid
+      - role: control-plane-invalid
+      - role: control-plane-invalid
```

Comment:
Shall we need the above 3 duplicated lines ?

Reply:
We have static IP field like following, in order to change all role names I repeated them
```  
    hosts:
    - role: bootstrap
      networkDevice:
        ipAddrs:
        - 192.168.221.120/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: control-plane
      # failureDomain: us-east-1a
      networkDevice:
        ipAddrs:
        - 192.168.221.121/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: control-plane
      # failureDomain: us-east-1b
      networkDevice:
        ipAddrs:
        - 192.168.221.122/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: control-plane
      # failureDomain: us-east-1c
      networkDevice:
        ipAddrs:
        - 192.168.221.123/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: compute
      networkDevice:
        ipAddrs:
        - 192.168.221.124/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
    - role: compute
      networkDevice:
        ipAddrs:
        - 192.168.221.125/24
        gateway: 192.168.221.1
        nameservers:
        - 192.168.221.1
```

Reply:
got it.